### PR TITLE
Identifiable composite nodes

### LINKED ISSUE #54: Subtasks with name
**Is your feature request related to a problem? Please describe.**
The current event log contains details about processed subtasks node in the form:

| Alias | Transition | Node Log |
|---|---|---|
| A1 | T1 | L1 |
| A2 | T2 | L2 |
| "composite" | | |

**Describe the solution you'd like**
I would like to name subtasks node:
```
node {
  factory = subtasks
  config {
    alias = subgraphA
    subtasks = [
      { 
        # subtask rootNodeDefinition
      },
      ...
    ]
  }
}
```
And in the event log see:

| Alias | Transition | Node Log |
|---|---|---|
| A1 | T1 | L1 |
| A2 | T2 | L2 |
| "subgraphA" | | |

**Describe alternatives you've considered**
Change log representation to graph.

### PR DESCRIPTION
**Pull request by tomaszmichalak**
Fixes #54.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: handler/engine/README.md [hunk 1]
```diff
@@ -2,48 +2,56 @@
...
+### Transition
+A directed graph consists of nodes and edges. Edges are called transitions. Transition is a simple text. 
+
+The pre-defined transitions are:
+- `_success` - the default one, indicates that operation completes successfully (no exception)
+- `_error` - means that operation has throw an exception
+
+There are two important rules to remember:
+> If a node responds with *_success* transition, but the transition is not configured, then 
+>processing is finished.
```
Comment (Mateusz512) [inline handler/engine/README.md]:
```suggestion
>processing of the graph/subgraph is finished.
```

File: handler/engine/README.md [hunk 2]
```diff
@@ -2,48 +2,56 @@
...
+
+> If a node responds with *_error* transition, but the transition is not configured, then an
```
Comment (Mateusz512) [inline handler/engine/README.md]:
```suggestion
> If a node responds with *_error* transition, but the `_error` transition is not configured, then an 
```

File: handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java [hunk 3]
```diff
@@ -81,14 +79,14 @@ void expectSuccessParallelProcessing(VertxTestContext testContext, Vertx vertx)
     JsonObject taskBPayload = new JsonObject().put("key", "taskBOperation");
     JsonObject taskCPayload = new JsonObject().put("key", "taskCOperation");
 
-    Node rootNode = new ActionNode("first", appendBody(":first"),
-        successTransition(new CompositeNode(
+    Node rootNode = new SingleNode("first", appendBody(":first"),
+        successTransition(new CompositeNode("composite",
```
Comment (Mateusz512) [inline handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java]:
```suggestion
        successTransition(new CompositeNode(COMPOSITE_NODE_ID,
```

File: handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java [hunk 4]
```diff
@@ -119,21 +117,21 @@ void expectSuccessParallelProcessing(VertxTestContext testContext, Vertx vertx)
   @DisplayName("Expect body updated after complex processing")
   void expectSuccessMultipleParallel(VertxTestContext testContext, Vertx vertx) throws Throwable {
     // given
-    Node rootNode = new ActionNode("first", success(),
-        successTransition(new CompositeNode(
+    Node rootNode = new SingleNode("first", success(),
+        successTransition(new CompositeNode("composite",
```
Comment (Mateusz512) [inline handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java]:
```suggestion
        successTransition(new CompositeNode(COMPOSITE_NODE_ID,
```

File: handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java [hunk 5]
```diff
@@ -119,21 +117,21 @@ void expectSuccessParallelProcessing(VertxTestContext testContext, Vertx vertx)
...
                 parallel(
-                    new ActionNode("A", appendPayload("A", ":payloadA"), NO_TRANSITIONS),
-                    new ActionNode("B", appendPayload("B", ":payloadB"), NO_TRANSITIONS)
+                    new SingleNode("A", appendPayload("A", ":payloadA"), NO_TRANSITIONS),
+                    new SingleNode("B", appendPayload("B", ":payloadB"), NO_TRANSITIONS)
                 ),
-                new ActionNode("middle", success(),
-                    successTransition(new CompositeNode(
+                new SingleNode("middle", success(),
+                    successTransition(new CompositeNode("composite",
```
Comment (Mateusz512) [inline handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java]:
```suggestion
                    successTransition(new CompositeNode(COMPOSITE_NODE_ID,
```

File: handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java [hunk 6]
```diff
@@ -159,26 +157,26 @@ void expectSuccessMultipleParallel(VertxTestContext testContext, Vertx vertx) th
   @DisplayName("Expect logs in order after complex processing")
   void expectProcessingLogsInOrder(VertxTestContext testContext, Vertx vertx) throws Throwable {
     // given
-    Node rootNode = new ActionNode("first", success(),
-        successTransition(new CompositeNode(
+    Node rootNode = new SingleNode("first", success(),
+        successTransition(new CompositeNode("composite",
```
Comment (Mateusz512) [inline handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java]:
```suggestion
        successTransition(new CompositeNode(COMPOSITE_NODE_ID,
```

File: handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java [hunk 7]
```diff
@@ -197,12 +195,12 @@ void expectProcessingLogsInOrder(VertxTestContext testContext, Vertx vertx) thro
               Operation.range("task", "A2", "ERROR", 1, 4),
               Operation.range("task", "A3-fallback", "SUCCESS", 1, 4),
               Operation.range("task", "B", "SUCCESS", 1, 4),
-              Operation.exact("task", COMPOSITE_NODE_ID, "SUCCESS", 5),
+              Operation.exact("task", "composite", "SUCCESS", 5),
```
Comment (Mateusz512) [inline handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java]:
```suggestion
              Operation.exact("task", COMPOSITE_NODE_ID, "SUCCESS", 5),
```

File: handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java [hunk 8]
```diff
@@ -197,12 +195,12 @@ void expectProcessingLogsInOrder(VertxTestContext testContext, Vertx vertx) thro
...
               Operation.exact("task", "middle", "SUCCESS", 6),
               Operation.range("task", "X", "SUCCESS", 7, 9),
               Operation.range("task", "Y1", "SUCCESS", 7, 9),
               Operation.range("task", "Y2", "SUCCESS", 7, 9),
-              Operation.exact("task", COMPOSITE_NODE_ID, "SUCCESS", 10),
+              Operation.exact("task", "composite", "SUCCESS", 10),
```
Comment (Mateusz512) [inline handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java]:
```suggestion
              Operation.exact("task", COMPOSITE_NODE_ID, "SUCCESS", 10),
```

File: handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java [hunk 9]
```diff
@@ -227,14 +225,14 @@ void ensureBodyImmutableDuringParallelProcessing(VertxTestContext testContext, V
   @DisplayName("Expect parallel nodes processed in parallel when delays")
   void verifyParallelExecution(VertxTestContext testContext, Vertx vertx) throws Throwable {
     // given
-    Node rootNode = new ActionNode("first", success(),
-        successTransition(new CompositeNode(
+    Node rootNode = new SingleNode("first", success(),
+        successTransition(new CompositeNode("composite",
```
Comment (Mateusz512) [inline handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java]:
```suggestion
        successTransition(new CompositeNode(COMPOSITE_NODE_ID,
```

File: handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java [hunk 10]
```diff
@@ -256,23 +254,23 @@ void verifyParallelExecution(VertxTestContext testContext, Vertx vertx) throws T
   @DisplayName("Expect nested parallel nodes processed in parallel when delays")
   void verifyNestedParallelExecution(VertxTestContext testContext, Vertx vertx) throws Throwable {
     // given
-    Node rootNode = new ActionNode("first", success(),
-        successTransition(new CompositeNode(
+    Node rootNode = new SingleNode("first", success(),
+        successTransition(new CompositeNode("composite",
```
Comment (Mateusz512) [inline handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java]:
```suggestion
        successTransition(new CompositeNode(COMPOSITE_NODE_ID,
```

File: handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java [hunk 11]
```diff
@@ -256,23 +254,23 @@ void verifyParallelExecution(VertxTestContext testContext, Vertx vertx) throws T
...
                 parallel(
-                    new ActionNode("A", successWithDelay(500), NO_TRANSITIONS),
-                    new CompositeNode(
+                    new SingleNode("A", successWithDelay(500), NO_TRANSITIONS),
+                    new CompositeNode("composite",
```
Comment (Mateusz512) [inline handler/engine/src/test/java/io/knotx/fragments/engine/TaskEngineScenariosTest.java]:
```suggestion
                    new CompositeNode(COMPOSITE_NODE_ID,
```

File: handler/engine/src/main/java/io/knotx/fragments/engine/TaskEngine.java [hunk 12]
```diff
@@ -59,14 +60,14 @@
   }
 
   private Single<FragmentResult> getResult(TaskExecutionContext context) {
-    return context.getCurrentNode().isComposite()
+    return NodeType.COMPOSITE.equals(context.getCurrentNode().getType())
```
Comment (marcinkp) [inline handler/engine/src/main/java/io/knotx/fragments/engine/TaskEngine.java]: for enums rather
`context.getCurrentNode().getType() == COMPOSITE`

File: handler/core/src/main/java/io/knotx/fragments/task/ConfigurationTaskProvider.java [hunk 13]
```diff
@@ -82,7 +82,12 @@ private Node buildCompositeNode(GraphNodeOptions options, Map<String, Node> edge
     List<Node> nodes = config.getSubtasks().stream()
         .map(this::initGraphRootNode)
         .collect(Collectors.toList());
-    return new CompositeNode("composite", nodes, edges.get(SUCCESS_TRANSITION), edges.get(ERROR_TRANSITION));
+    return new CompositeNode(getNodeId(), nodes, edges.get(SUCCESS_TRANSITION), edges.get(ERROR_TRANSITION));
+  }
+
+  private String getNodeId() {
+    // TODO this value should be calculated based on graph, the behaviour now is not changed
```
Comment (marcinkp) [inline handler/core/src/main/java/io/knotx/fragments/task/ConfigurationTaskProvider.java]: do we really need it here?

**Pull request merged**